src/connect/components/drive-view/drive-view.tsx

```diff
@@ -1,189 +1,82 @@
-    CreateDriveModal,
-    DriveTreeItem,
-    DriveType,
-    FOLDER,
-    LOCAL,
-    LOCAL_DRIVE,
-    PUBLIC,
-    PUBLIC_DRIVE,
-    SHARED,
-    TreeItemType,
-    usePathContent,
+    DragAndDropProps,
+    NodeProps,
+    SharingType,
+    TUiNodesContext,
+    UiDriveNode,
 } from '@/connect';
 import { Icon } from '@/powerhouse';
-import { useState } from 'react';
+import { ReactNode } from 'react';
 import { twJoin, twMerge } from 'tailwind-merge';
 
-export interface DriveViewProps
-    extends Omit<
-        React.HTMLAttributes<HTMLDivElement>,
-        'onDragEnd' | 'onDragStart'
-    > {
-    type: DriveType;
-    name: string;
-    defaultItemOptions?: ConnectTreeViewItemProps['defaultOptions'];
-    drivePath?: string;
-    disableHighlightStyles?: boolean;
-
-    isAllowedToCreateDocuments?: boolean;
-    onDropEvent?: ConnectTreeViewProps['onDropEvent'];
-    onItemClick?: ConnectTreeViewProps['onItemClick'];
-    onSubmitInput?: ConnectTreeViewProps['onSubmitInput'];
-    onCancelInput?: ConnectTreeViewProps['onCancelInput'];
-    onItemOptionsClick?: ConnectTreeViewProps['onItemOptionsClick'];
-    onDropActivate?: ConnectTreeViewProps['onDropActivate'];
-    onDragStart?: ConnectTreeViewProps['onDragStart'];
-    onDragEnd?: ConnectTreeViewProps['onDragEnd'];
-    onCreateDrive?: (drive: AddDriveInput | AddPublicDriveInput) => void;
-    disableAddDrives?: boolean;
-    displaySyncFolderIcons?: boolean;
-}
-
-const filterDriveByType = (drive: DriveTreeItem, type: DriveType) => {
-    switch (type) {
-        case 'PUBLIC_DRIVE':
-            return drive.type === 'PUBLIC_DRIVE';
-        case 'LOCAL_DRIVE':
-            return drive.type === 'LOCAL_DRIVE';
-        case 'CLOUD_DRIVE':
-            return drive.type === 'CLOUD_DRIVE';
-        default:
-            return false;
-    }
-};
+export type DriveViewProps = TUiNodesContext &
+    NodeProps &
+    DragAndDropProps & {
+        label: ReactNode;
+        groupSharingType: SharingType;
+        disableAddDrives: boolean;
+        className?: string;
+        showAddDriveModal: (groupSharingType: SharingType) => void;
+        showDriveSettingsModal: (uiDriveNode: UiDriveNode) => void;
+    };
 
 export function DriveView(props: DriveViewProps) {
     const {
+        driveNodes,
+        selectedDriveNode,
+        label,
+        groupSharingType,
         className,
-        type,
-        name,
-        onDropEvent,
-        onItemClick,
-        onSubmitInput,
-        onItemOptionsClick,
-        defaultItemOptions,
-        onDropActivate,
-        onDragStart,
-        onDragEnd,
-        onCancelInput,
-        disableHighlightStyles,
-        drivePath = '/',
-        onCreateDrive,
         disableAddDrives,
-        isAllowedToCreateDocuments = true,
-        displaySyncFolderIcons = false,
-        ...restProps
+        isAllowedToCreateDocuments,
+        showAddDriveModal,
     } = props;
-    const [showAddModal, setShowAddModal] = useState(false);
```

Comment:
Control of modal showing has been moved to the connect app, to be consistent with the other modal types.